### PR TITLE
[INFRA/RULES] Added RulesToPlay to gamestate

### PR DESCRIPTION
**Pull request by Neelesh99**
# Summary

As title.

### CODE REVIEW

**Comment by QFSW** [comment]
# ⚠️Warning! This diff contains changes in protected folders

File: internal/common/gamestate/gamestate.go [hunk 1]
```diff
@@ -114,6 +120,41 @@ func copyIIGOHistory(iigoHistory map[uint][]shared.Accountability) map[uint][]sh
...
+	return rules.RuleLink{
+		Linked:     inp.Linked,
+		LinkType:   inp.LinkType,
+		LinkedRule: inp.LinkedRule,
+	}
+}
+
+func copyRequiredVariables(inp []rules.VariableFieldName) []rules.VariableFieldName {
+	targetList := make([]rules.VariableFieldName, len(inp))
+	for index, entry := range inp {
```
Comment (lhl2617) [inline internal/common/gamestate/gamestate.go]: as per linter :) array copying is built in with `copy(dst, src)`

File: internal/server/server.go [hunk 2]
```diff
@@ -2,14 +2,14 @@ package server
 
 import (
 	"fmt"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/rules"
```
Comment (lhl2617) [inline internal/server/server.go]: linter

File: internal/server/server.go [hunk 3]
```diff
@@ -59,6 +59,12 @@ func createSOMASServer(
 		forageHistory[t] = make([]foraging.ForagingReport, 0)
 	}
 
+	if gameConfig.IIGOConfig.StartWithRulesInPlay {
+		for ruleName := range rules.AvailableRules {
+			_ = rules.PullRuleIntoPlay(ruleName)
```
Comment (lhl2617) [inline internal/server/server.go:66]: can you add a comment as to why this err is ignored?

**Review by lhl2617 — APPROVED** [review]
pre-approving, looks good

File: internal/server/server.go (hunk 3 above)
Comment (Neelesh99) [reply internal/server/server.go:66]: Yup, thanks

Comment (lhl2617) [reply internal/server/server.go:66]: I personally still wouldn't ignore btw--at least log it out in case a bug happens so it's easier to debug

**Pull request merged**